src/main/java/com/laserfiche/repository/api/RepositoryApiClientImpl.java

```diff
@@ -48,13 +55,19 @@ protected RepositoryApiClientImpl(String servicePrincipalKey, AccessKey accessKe
         templateDefinitionsClient = new TemplateDefinitionsClientImpl(baseUrl, httpClient);
     }
 
-    public static RepositoryApiClient CreateFromAccessKey(String servicePrincipalKey, AccessKey accessKey,
+    public static RepositoryApiClient createFromAccessKey(String servicePrincipalKey, AccessKey accessKey,
             String baseUrlDebug) {
-        return new RepositoryApiClientImpl(servicePrincipalKey, accessKey, baseUrlDebug);
+        return new RepositoryApiClientImpl(servicePrincipalKey, accessKey, null, null, null, null, baseUrlDebug);
+    }
+
+    public static RepositoryApiClient createFromAccessKey(String servicePrincipalKey, AccessKey accessKey) {
+        return createFromAccessKey(servicePrincipalKey, accessKey, null);
     }
 
-    public static RepositoryApiClient CreateFromAccessKey(String servicePrincipalKey, AccessKey accessKey) {
-        return CreateFromAccessKey(servicePrincipalKey, accessKey, null);
+    public static RepositoryApiClient createFromUsernamePassword(String repoId, String username, String password,
+            String baseUrl) {
+        String baseUrlWithSlash = baseUrl.endsWith("/") ? baseUrl.substring(0, baseUrl.length() - 1) : baseUrl;
```

Comment:
could move this line that trims the baseUrl to `RepositoryApiClientImpl` since it also has the baseUrl parameter and if `createFromAccessKey(String servicePrincipalKey, AccessKey accessKey, String baseUrlDebug)` is called, the trim functionality would be reused

Reply:
ok done